### ISSUE TITLE
How to directly define the LayerNorm module?

### ISSUE THREAD

**Issue by hobbitlzy** [facebookresearch/CrypTen#399]
I can use the function of LayerNorm using `crypten.nn.from_pytorch()` from an existing torch model.  But I find there is no corresponding `crypten.nn.LayerNorm` defined. What if I want to directly define a crypted LayerNorm module?

**Comment by lvdmaaten** [comment]
You would need to 1 to 4 things to add this new module:
1. Implement a `LayerNorm` module in CrypTen. [This](https://github.com/facebookresearch/CrypTen/blob/main/crypten/nn/module.py#L2748) implements `BatchNormalization` and should give you a sense of how to do this. Note that CrypTen modules have a static `from_onnx` function that describes how attributes map to module hyperparameters (probably not relevant here). Apart from that, we try to match the PyTorch module signature as closely as possible.
2. If you need to implement a new function (and its gradient) to enable to `LayerNorm` module, you can do so in `gradients.py`. [This](https://github.com/facebookresearch/CrypTen/blob/main/crypten/gradients.py#L1761) shows the implementation of the `batchnorm` function. We try to match the PyTorch function signature as closely as possible. Note that the `@register_function` decorator works anywhere, so you can also add functions to CrypTen's autograd in user code without making changes in CrypTen itself.
3. If the ONNX op name does not match the CrypTen/PyTorch `Module` name, you can manually add the mapping from the ONNX op name to the correct `Module` [in the `ONNX_TO_CRYPTEN` dict](https://github.com/facebookresearch/CrypTen/blob/main/crypten/nn/onnx_converter.py#L139).
4. Add unit tests for each component you're adding.

Feel free to submit a PR -- `LayerNorm` seems like it would be useful to have!

**Comment by hobbitlzy** [comment]
Thanks for your reply😁, I will manage to do this. But I still have a question. If this module is not  implemented yet, why was I able to  transform a model with `LayerNorm` sub-module using `crypten.nn.from_pytorch()` ?
